lib/babe/build.go

```diff
@@ -347,16 +348,17 @@ func hasSlotEnded(slot Slot) bool {
 	return time.Since(slotEnd) >= 0
 }
 
-// ExtrinsicsToBody returns scale encoded block body which contains inherent and extrinsic.
+// ExtrinsicsToBody returns scale2 encoded block body which contains inherent and extrinsic.
 func ExtrinsicsToBody(inherents [][]byte, txs []*transaction.ValidTransaction) (*types.Body, error) {
 	extrinsics := types.BytesArrayToExtrinsics(inherents)
 
 	for _, tx := range txs {
-		decExt, err := scale.Decode(tx.Extrinsic, []byte{})
+		var decExt []byte
+		err := scale.Unmarshal(tx.Extrinsic, &decExt)
 		if err != nil {
 			return nil, err
 		}
-		extrinsics = append(extrinsics, decExt.([]byte))
+		extrinsics = append(extrinsics, decExt)
 	}
 
 	return types.NewBodyFromExtrinsics(extrinsics)
```

Comment:
You should be able to unmarshal the whole `types.Body` using the scale package.  `NewBodyFromExtrinsics` is using the old package.

Reply:
Okay I will look into doing this